src/meta/processors/admin/Balancer.cpp

```diff
@@ -313,6 +320,77 @@ StatusOr<HostAddr> Balancer::hostWithMinimalParts(
     return Status::Error("No host is suitable for %d", partId);
 }
 
+folly::Future<Status> Balancer::leaderBalance() {
+    folly::Promise<Status> promise;
+    auto future = promise.getFuture();
+
+    std::vector<GraphSpaceID> spaces;
+    kvstore::ResultCode ret = kvstore::ResultCode::SUCCEEDED;
+    if (!getAllSpaces(spaces, ret)) {
+        return Status::Error("Can't access kvstore, ret = %d", static_cast<int32_t>(ret));
+    }
+
+    bool expected = false;
+    if (running_.compare_exchange_strong(expected, true)) {
+        hostLeaderMap_.reset(new HostLeaderMap);
+        client_->getLeaderDist(hostLeaderMap_.get())
+            .then([p = std::move(promise), allSpaces = std::move(spaces), this]
+                  (folly::Try<Status>&& t) mutable {
+            if (t.hasException()) {
+                p.setValue(Status::Error(folly::stringPrintf("RPC failure in AdminClient: %s",
+                                                             t.exception().what().c_str())));
+                return;
+            }
+            if (!t.value().ok()) {
+                p.setValue(t.value());
+                return;
+            }
+
+            LeaderBalancePlan plan;
+            for (const auto& space : allSpaces) {
+                buildLeaderBalancePlan(hostLeaderMap_.get(), space, plan);
```

Comment:
hostLeaderMap_ is member of this, no need pass by parameter.

Reply:
Sure.